tools/generate-hooks-docs.php

```
@@ -322,11 +403,308 @@ static function ( $a, $b ) {
 
+/**
+ * Find or generate an example usage for a hook.
+ * 
+ * @param string $hook       The hook name.
+ * @param string $type       Hook type ('action' or 'filter').
+ * @param array  $params     Parameter information array.
+ * @param array  $return_value Return value information (for filters).
+ * @param string $docblock   Raw docblock content if available.
+ * @return string Example usage as markdown code block.
+ */
+function edac_generate_hook_example( $hook, $type, $params, $return_value, $docblock = '' ) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$return_value'. (undefined)

(UnusedFormalParameter)